### PR TITLE
vim-patch:8.2.4188: not all gitconfig files are recognized

### PR DESCRIPTION
**Pull request by clason**
Problem:    Not all gitconfig files are recognized.
Solution:   Add a few more patterns. (Tim Pope, closes vim/vim#9597)
https://github.com/vim/vim/commit/bcfa11b7dfdfbb4d412dd843a6da3fce68ba2e39

### CODE REVIEW

File: runtime/lua/vim/filetype.lua [hunk 1]
```diff
@@ -892,7 +892,6 @@ local filename = {
   [".gitconfig"] = "gitconfig",
   [".gitmodules"] = "gitconfig",
   ["/.config/git/config"] = "gitconfig",
-  ["/etc/gitconfig"] = "gitconfig",
```
Comment (gpanders) [inline runtime/lua/vim/filetype.lua:895]: This looks correct but the line just above this looks suspicious. I think that should also be moved to a pattern style entry with `.*` at the beginning. Are you okay doing that as part of this PR or do you want to do it separately?

File: runtime/lua/vim/filetype.lua [hunk 2]
```diff
@@ -1372,6 +1370,10 @@ local pattern = {
       return "git"
     end
   end,
+  [".*/etc/gitconfig"] = "gitconfig",
+  [".*/.config/git/config"] = "gitconfig",
```
Comment (gpanders) [inline runtime/lua/vim/filetype.lua:1374]: `.` here needs to be escaped

```suggestion
  [".*/%.config/git/config"] = "gitconfig",
```

Comment (clason) [reply runtime/lua/vim/filetype.lua]: D'oh!

**Review by gpanders — APPROVED** [review]

File: runtime/lua/vim/filetype.lua (hunk 2 above)
Comment (clason) [reply runtime/lua/vim/filetype.lua]: Turns out that was already part of the original `filetype.lua`...

(We do need to come up with a way of sorting these...)

**Review by gpanders — APPROVED** [review]
Good catch

**Comment by clason** [comment]
wasn't me, was lint :]

**Pull request merged**